### DISCUSSION TITLE
Does Redis auth works through envFromSecret ?

### DISCUSSION THREAD

**Discussion by vipmarwah** [apache/superset#25240] [General]
I am using Superset helm chart 0.10.6 and defined secrets for custom postgres and redis server as below

superset-secrets:
    apiVersion: v1
    kind: Secret
    metadata:
      name: superset-secrets
      namespace: mysuperset
    type: Opaque
    stringData:
      REDIS_HOST: redis-replication
      REDIS_PASSWORD: password
      REDIS_PORT: 6379
      DB_HOST: cnpg-rw
      DB_PORT: 5432
      DB_USER: postgres
      DB_PASS: superset
      DB_NAME: superset
---------------------------------------------------
I applied the secret in values.yaml as below

envFromSecret: superset-secrets
------------------------------------------------
Getting error:
ERROR/MainProcess] consumer: Cannot connect to redis://redis-replication:6379/0: Authentication required..

Below is the screenshot of superset_config.py. It received all the values except redis_password

![image](https://github.com/apache/superset/assets/101658736/75544a1d-fd53-4e8f-aeae-17e6b909ad64)

Is there any other way to pass redis_password through secret? any help will be appreciated
------------------------------------------------------------------

Its works fine when we pass the redis_password directly in values.yaml file.

**Comment by v-em** [comment]
You can use extraEnvRaw in values.yaml:
```
extraEnvRaw:
  - name: REDIS_PASS
    valueFrom: 
      secretKeyRef:
        name: redis-password-secret
        key: redis-password

**Comment by YhebBOUSSELMI** [comment]
I ran into this issue and solved it by overriding the _superset-config_  helper defined in the superset [_helpers.tpl file](https://github.com/apache/superset/blob/21f85a414511cf298b6f6531792b846e59926a56/helm/superset/templates/_helpers.tpl#L65) 
**The Problem**:
When the Redis password is provided via a Kubernetes secret (rather than directly in values.yaml), the Helm template condition {{- if .Values.supersetNode.connections.redis_password }} evaluates to false. This causes the password authentication to be excluded from the generated config, even though the password will exist at runtime as an environment variable.

**The Solution**:
I moved the password check from Helm template logic to Python runtime logic. Instead of conditionally including the password field during template rendering, which always lets Python check if the REDIS_PASSWORD environment variable exists.

### Change 1: Redis Base URL
I replaced this:

```
# Redis Base URL
{{- if .Values.supersetNode.connections.redis_password }}
REDIS_BASE_URL=f"{env('REDIS_PROTO')}://{env('REDIS_USER', '')}:{env('REDIS_PASSWORD')}@{env('REDIS_HOST')}:{env('REDIS_PORT')}"
{{- else }}
REDIS_BASE_URL=f"{env('REDIS_PROTO')}://{env('REDIS_HOST')}:{env('REDIS_PORT')}"
{{- end }}
```

With this: 
```
# Redis Base URL
REDIS_USER = env('REDIS_USER', '')
REDIS_PASSWORD = env('REDIS_PASSWORD', '')

if REDIS_PASSWORD:
    REDIS_BASE_URL = f"{env('REDIS_PROTO')}://{REDIS_USER}:{REDIS_PASSWORD}@{env('REDIS_HOST')}:{env('REDIS_PORT')}"
else:
    REDIS_BASE_URL = f"{env('REDIS_PROTO')}://{env('REDIS_HOST')}:{env('REDIS_PORT')}"
```    
    
### Change 2: RESULTS_BACKEND
I replaced this: 
```
RESULTS_BACKEND = RedisCache(
      host=env('REDIS_HOST'),
      {{- if .Values.supersetNode.connections.redis_password }}
      password=env('REDIS_PASSWORD'),
      {{- end }}
      port=env('REDIS_PORT'),
      key_prefix='superset_results',
      {{- if .Values.supersetNode.connections.redis_ssl.enabled }}
      ssl=True,
      ssl_cert_reqs=env('REDIS_SSL_CERT_REQS'),
      {{- end }}
)
```

With this:     
```
# Build Redis cache config dynamically
redis_cache_config = {
    'host': env('REDIS_HOST'),
    'port': env('REDIS_PORT'),
    'key_prefix': 'superset_results',
}

# Add password if present
if env('REDIS_PASSWORD'):
    redis_cache_config['password'] = env('REDIS_PASSWORD')

# Add SSL config if cert reqs is set
if env('REDIS_SSL_CERT_REQS'):
    redis_cache_config['ssl'] = True
    redis_cache_config['ssl_cert_reqs'] = env('REDIS_SSL_CERT_REQS')

RESULTS_BACKEND = RedisCache(**redis_cache_config)
```

I hope this helps! 
Cheers!

**Reply by bxsx** [reply]
Hi @YhebBOUSSELMI No need to patch anything in your case. You just need to set a dummy password in your `values.yaml`, for example: `supersetNode.connections.redis_password = dummy-password`. This will make the condition in the template evaluate to `true`, and the password from your secret will be correctly picked up and used. Cheers
